src/include/daos/object.h

```diff
@@ -341,37 +365,34 @@ daos_oclass_is_ec(struct daos_oclass_attr *oca)
 }
 
 static inline void
-daos_obj_set_oid(daos_obj_id_t *oid, daos_ofeat_t ofeats,
-		 daos_oclass_id_t cid, uint32_t args)
+daos_obj_set_oid(daos_obj_id_t *oid, enum daos_obj_type type,
+		 enum daos_obj_redun ord, uint32_t nr_grps,
+		 uint32_t args)
 {
 	uint64_t hdr;
 
-	/* TODO: add check at here, it should return error if user specified
-	 * bits reserved by DAOS
-	 */
+	/** XXX: encode nr_grps as-is for now */
+
 	oid->hi &= (1ULL << OID_FMT_INTR_BITS) - 1;
 	/**
 	 * | Upper bits contain
-	 * | OID_FMT_VER_BITS (version)		 |
-	 * | OID_FMT_FEAT_BITS (object features) |
+	 * | OID_FMT_TYPE_BITS (object features) |
 	 * | OID_FMT_CLASS_BITS (object class)	 |
-	 * | 96-bit for upper layer ...		 |
+	 * | OID_FMT_MD_BITS (object metadata)	 |
+	 * | 96-bit for API user ...		 |
 	 */
-	hdr  = ((uint64_t)OID_FMT_VER << OID_FMT_VER_SHIFT);
-	hdr |= ((uint64_t)ofeats << OID_FMT_FEAT_SHIFT);
-	hdr |= ((uint64_t)cid << OID_FMT_CLASS_SHIFT);
+	hdr  = ((uint64_t)type << OID_FMT_TYPE_SHIFT);
+	hdr |= ((uint64_t)ord << OID_FMT_CLASS_SHIFT);
+	hdr |= ((uint64_t)nr_grps << OID_FMT_MD_SHIFT);
 	oid->hi |= hdr;
```

Comment:
I have only change the packing of the OID for now. The unpacking and extracting the class and number of groups has to be fixed too.